google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineApplicationInfoPanel.java

```diff
@@ -67,10 +67,22 @@ public AppEngineApplicationInfoPanel() {
   }
 
   @SuppressWarnings("FutureReturnValueIgnored")
+  /**
+   * Updates the panel as follows:
+   *   if the project textbox specifies a valid project, it displays the project's information,
+   *   if the project textbox specifies an invalid project, it displays an error message,
+   *   if the project textbox is empty, no message is displayed.
+   */
   public void refresh(final ProjectSelectionChangedEvent event) {
     if (event == null) {
       ApplicationManager.getApplication().executeOnPooledThread(
-          () -> setMessage(GctBundle.getString("appengine.infopanel.noproject"),
+          () -> setMessage("", false /* isError*/));
```

Comment:
If we need to clear out the message explicitly like this, can you add a `clearMessage` (or something like that) method for better readability?

Reply:
Done